socketserver.h

```diff
@@ -110,7 +112,7 @@ class SocketServer
         _cbReceived(0)
     {
         _abOutputBuffer = std::make_unique<uint8_t []>(_maximumPacketSize);        // Must add +1 for LZ bug case in Arduino library if ever backported there
-        _pBuffer = std::make_unique<uint8_t []>(_maximumPacketSize);
+        _pBuffer        = std::make_unique<uint8_t []>(_maximumPacketSize);
         memset(&_address, 0, sizeof(_address));

```

Comment:
nit: you could probably move all of these into the initialization block:
```C++
    SocketServer(int port, size_t maxLEDs) :
        _port(port),
        _server_fd(-1),
        _address{},
        _maximumPacketSize(STANDARD_DATA_HEADER_SIZE + LED_DATA_SIZE * maxLEDs),       
        _abOutputBuffer(std::make_unique<uint8_t []>(_maximumPacketSize),
        _pBuffer = std::make_unique<uint8_t []>(_maximumPacketSize),
        _cbReceived(0)
       {}
```
Compilers make slightly better code in such cases (and you won't need to `memset` _address below)

Note that you'll need to move `_maximumPacketSize` to be declared before both buffers for this to work

Reply:
Also you could make some of that initialization inline
```C++
class SocketServer
{
    int                         _port;
    int                         _server_fd{-1};
    struct sockaddr_in          _address{};
    size_t _maximumPacketSize;
    std::unique_ptr<uint8_t []> _pBuffer;
    std::unique_ptr<uint8_t []> _abOutputBuffer;
public:
    size_t cbReceived{0};
public:
     SocketServer(int port, size_t maxLEDs) :
        _maximumPacketSize(STANDARD_DATA_HEADER_SIZE + LED_DATA_SIZE * maxLEDs),
       _abOutputBuffer(std::make_unique<uint8_t []>(_maximumPacketSize),
       _pBuffer = std::make_unique<uint8_t []>(_maximumPacketSize)
       {}
};
```